packages/storage-driver-s3/src/index.ts

```diff
@@ -39,12 +39,16 @@ export class DriverS3 implements Driver {
 	private serverSideEncryption: string | undefined;
 
 	constructor(config: DriverS3Config) {
-		const s3ClientConfig: S3ClientConfig = {
-			credentials: {
+		const s3ClientConfig: S3ClientConfig = {};
+
+		if (config.key || config.secret) {
+			if (!(config.key && config.secret)) throw new Error('Both `key` and `secret` are required when defined');
```

Comment:
```suggestion
		if (isEmpty(config.key) || isEmpty(config.secret)) {
                    throw new Error('Both `key` and `secret` are required when defined');
                } else {
```
If i got my boolean comparisons right this would be a more readable option in my opinion.

Reply:
Actually i made a mistake with the usage of isEmpty here.